packages/pyright-internal/src/common/commandLineOptions.ts

```diff
@@ -145,14 +136,35 @@ export class CommandLineOptions {
     // Run ambient analysis.
     enableAmbientAnalysis = true;
 
-    // Analyze functions and methods that have no type annotations?
-    analyzeUnannotatedFunctions?: boolean;
-
     // Disable reporting of hint diagnostics with tags?
     disableTaggedHints?: boolean;
+}
+
+// Some options can be specified from a source other than the pyright config file.
+// This can be from command-line parameters or some other settings mechanism, like
+// that provided through a language client like the VS Code editor. These options
+// are later combined with those from the config file to produce the final configuration.
+export class CommandLineOptions {
+    // Settings that are possible to set in a config.json file.
+    configSettings: CommandLineConfigOptions = new CommandLineConfigOptions();
```

Comment:
I'm not super happy with this name. config is used so much here. 'environmentSettings' maybe? As that's what the configuration.md calls them mostly now. Although it also includes type checking settings :(

Reply:
Hmm, "configSettings" seems good to me.

Reply:
typeCheckingSettings?

Reply:
Okay I'll just leave it.